### ISSUE TITLE
Results are bad/random

### ISSUE THREAD

**Issue by KoenL-1747481** [akTwelve/tutorials#18]
I have tried the demo file of your fork of the matterport mask rcnn repo and the resulting segmentations are very random :
![image](https://user-images.githubusercontent.com/61016468/154811124-dc0859a9-4d50-4bca-adc8-5dc70292a248.png)
I experience the same issue with this tutorial. I also tried "augmented startups'" fork of the matterport mask rcnn repo (https://github.com/augmentedstartups/Mask_RCNN#readme) but running that demo also gives very random segmentations. It seems like the COCO pre-trained weights are just not working or something. Any idea how to fix this?  Results from this tutorial look like this:  
![image](https://user-images.githubusercontent.com/61016468/154811252-bd1468e5-d785-487a-aad9-8fab8c043e9d.png)

I tried this on 2 pc's with installations from scratch and both have the same random results.

**Comment by akTwelve** [comment]
It’s probably just drawing them wrong. Sometimes X and Y or Width and Height get swapped and the segmentations don’t get drawn correctly. I haven’t used this repo in a long time so I’m not exactly sure what’s up though.

**Comment by m11005217** [comment]
> I have tried the demo file of your fork of the matterport mask rcnn repo and the resulting segmentations are very random : ![image](https://user-images.githubusercontent.com/61016468/154811124-dc0859a9-4d50-4bca-adc8-5dc70292a248.png) I experience the same issue with this tutorial. I also tried "augmented startups'" fork of the matterport mask rcnn repo (https://github.com/augmentedstartups/Mask_RCNN#readme) but running that demo also gives very random segmentations. It seems like the COCO pre-trained weights are just not working or something. Any idea how to fix this? Results from this tutorial look like this: ![image](https://user-images.githubusercontent.com/61016468/154811252-bd1468e5-d785-487a-aad9-8fab8c043e9d.png)
> 
> I tried this on 2 pc's with installations from scratch and both have the same random results.

Did you solve this problem,  I also have same problem.

**Comment by KoenL-1747481** [comment]
> > I have tried the demo file of your fork of the matterport mask rcnn repo and the resulting segmentations are very random : ![image](https://user-images.githubusercontent.com/61016468/154811124-dc0859a9-4d50-4bca-adc8-5dc70292a248.png) I experience the same issue with this tutorial. I also tried "augmented startups'" fork of the matterport mask rcnn repo (https://github.com/augmentedstartups/Mask_RCNN#readme) but running that demo also gives very random segmentations. It seems like the COCO pre-trained weights are just not working or something. Any idea how to fix this? Results from this tutorial look like this: ![image](https://user-images.githubusercontent.com/61016468/154811252-bd1468e5-d785-487a-aad9-8fab8c043e9d.png)
> > I tried this on 2 pc's with installations from scratch and both have the same random results.
> 
> Did you solve this problem, I also have same problem.

Yes and no.. i didn't fix it in this repo, but i stumbled upon facebook's mask rcnn implementation called detectron2 (search for the github repo). Its still actively being updated and it works perfectly with just a few lines of code (https://youtu.be/Pb3opEFP94U)

**Comment by m11005217** [comment]
Grateful ! Thanks for your help.
I will do my best

**Comment by ansacaron** [comment]
Exactly the same problem here... any clue?

**Comment by KoenL-1747481** [comment]
> Exactly the same problem here... any clue?

I moved on with facebook's mask rcnn implementation as i mentioned in a previous comment. Otherwise i have no clue

**Comment by pooyasohrabipoor** [comment]
Hi 
I am trying to use this code and I am running to the same issue even when I use the same code and dataset (cig

> It’s probably just drawing them wrong. Sometimes X and Y or Width and Height get swapped and the segmentations don’t get drawn correctly. I haven’t used this repo in a long time so I’m not exactly sure what’s up though.

**Comment by pooyasohrabipoor** [comment]
> It’s probably just drawing them wrong. Sometimes X and Y or Width and Height get swapped and the segmentations don’t get drawn correctly. I haven’t used this repo in a long time so I’m not exactly sure what’s up though.

Hi
I am trying to use your code and your dataset but not only dont I get a good inference result, but also the validation loss is way higher than your results ( around 2 after 8 epochs) and when I use the inference my results are so random and bad. I do not want to use some tools like mmdetection because later on I need to modify maskrcnn code for my dataset. Do you have any idea or suggestions?
I really appreciate your assistance.
